src/main/java/io/vertx/ext/mongo/impl/MongoClientImpl.java

```diff
@@ -962,11 +962,11 @@ public MongoClient distinct(String collection, String fieldName, String resultCl
 
   @Override
   public MongoClient distinctWithQuery(String collection, String fieldName, String resultClassname, JsonObject query, Handler<AsyncResult<JsonArray>> resultHandler) {
-    return distinctWithQuery(collection, fieldName, resultClassname, query, resultHandler, null);
+    return distinctWithQuery(collection, fieldName, resultClassname, query, null, resultHandler);
   }
 
   @Override
-  public MongoClient distinctWithQuery(String collection, String fieldName, String resultClassname, JsonObject query, Handler<AsyncResult<JsonArray>> resultHandler, DistinctOptions distinctOptions) {
+  public MongoClient distinctWithQuery(String collection, String fieldName, String resultClassname, JsonObject query, DistinctOptions distinctOptions, Handler<AsyncResult<JsonArray>> resultHandler) {
```

Comment:
This is a legitimate breaking change. We must document it in the breaking changes wiki page after merging.

Reply:
Shall I document it in the wiki and/or move it to another PR (like above's issue with `collation` vs `collationOptions`)?